rust/src/smb/smb.rs

```diff
@@ -790,7 +796,21 @@ impl SMBState {
         }
     }
 
+    fn state_stats (&mut self) {
+        let mut hm_memusage = hashmap_mem(&self.ssn2vec_map);
+        hm_memusage += hashmap_mem(&self.guid2name_map);
+        hm_memusage += hashmap_mem(&self.ssn2vecoffset_map);
+        hm_memusage += hashmap_mem(&self.ssn2tree_map);
+        hm_memusage += hashmap_mem(&self.ssnguid2vec_map);
+
+        let tx_mem = tx_queue_memory(&self.transactions);
+
+        SMBHASH_MEMORY.fetch_add(hm_memusage, Ordering::SeqCst);
```

Comment:
I never see this be decreased again, so what are we really tracking here?

Reply:
Good point. How do we track it more granularly? Recording how much was used when destroying the object gives a sense of the impact of the HashMap memory usage.